core/src/Streamly/Internal/Data/Array/Foreign/Mut/Type.hs

```diff
@@ -1849,6 +1849,27 @@ writeNWith alloc n = FL.take n (writeNWithUnsafe alloc n)
 writeN :: forall m a. (MonadIO m, Storable a) => Int -> Fold m a (Array a)
 writeN = writeNWith newArray
 
+-- /Pre-release/
+{-# INLINE writeRevN #-}
+writeRevN :: forall m a. (MonadIO m, Storable a) => Int -> Fold m a (Array a)
+writeRevN n = Fold step initial (return .fromArrayUnsafe . fst)
+    where
+
+    initial = do
+        arr <- newArray (max n 0)
+        return $ FL.Partial (toArrayUnsafe arr, n - 1)
```

Comment:
Do we need to store the count as we have start and end of the array? Having redundant/additional data in state causes performance issues. Also, we are not using a strict tuple here.

Reply:
this is the index where we poke the element in reverse order.